### ISSUE TITLE
Add run_gsg.py and run_gsg_no_trainer.py pre-training scripts to examples

### ISSUE THREAD

**Issue by KMFODA** [huggingface/transformers#20532]
### Feature request

Models such as Pegasus and LongT5 have been pretrained using Gap Sentences Generation (GSG) strategy rather the typical Masked Language Modelling (MLM). 

This pre-training strategy leads to improved performance in certain language tasks such as [summarisation](https://arxiv.org/pdf/1912.08777.pdf). This request is to add run_gsg_.py and run_gsg_no_trainer.py files in the examples folder that would enable pre-training using the GSG strategy instead of or on top of MLM.

### Motivation

This will enable users to pre-train Pegasus or LongT5 models from scratch or to continue pre-training existing checkpoints on new datasets.

### Your contribution

I've started thinking about how to build this and am happy to contribute a PR if the HF team think this is valuable and can offer advice on the best ways to approach this.

**Comment by sgugger** [comment]
We try to avoid having examples that are too specific in the maintained examples, as we don't have the bandwidth for too many of them. How about you host it in a repo of yourself and then link to it from the model pages in our doc as well as the community page?

**Comment by KMFODA** [comment]
Ah I see, not problem that seems like a good alternative. Where would be the best place for asking for help with road blocks if I stumble across any?

**Comment by sgugger** [comment]
You can use this issue or the [forums](https://discuss.huggingface.co/) :-)

**Comment by KMFODA** [comment]
Thank you @sgugger. I've had to de-prioritise this for due to funding constraints that will delay when we can train a bigger version of LongT5 from scratch so I'll close this for now and if we pick this back up I'll post any qs in the forums
.
